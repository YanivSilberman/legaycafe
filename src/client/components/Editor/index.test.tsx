import * as React from 'react';
import * as ReactShallowRenderer from 'react-test-renderer/shallow'
import Component from './index';

const defaultProps = {
  editorState: {},
  setEditorState: () => null,
  onFocus: () => null,
  onBlur: () => null,
  readOnly: false,
  text: "test",
  classes: {}
}

describe('Editor component test',() => {
  const renderer = ReactShallowRenderer.createRenderer();

  it('component loads for editing', () => {
    expect(renderer.render(
      <Component
        {...defaultProps}
      />
    )).toMatchSnapshot()
  })

  it('component loads read only', () => {
    expect(renderer.render(
      <Component
        readOnly
        text="test 123"
      />
    )).toMatchSnapshot()
  })

});
