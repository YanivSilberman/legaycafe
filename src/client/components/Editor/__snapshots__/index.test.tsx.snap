// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor component test component loads for editing 1`] = `
<LegayEditor
  classes={
    Object {
      "container": "LegayEditor-container-3",
      "editorContainer": "LegayEditor-editorContainer-5",
      "emojiSelect": "LegayEditor-emojiSelect-6",
      "footerSubsection": "LegayEditor-footerSubsection-2",
      "paper": "LegayEditor-paper-1",
      "readContainer": "LegayEditor-readContainer-4",
    }
  }
  editorState={Object {}}
  onBlur={[Function]}
  onFocus={[Function]}
  readOnly={false}
  setEditorState={[Function]}
  text="test"
/>
`;

exports[`Editor component test component loads read only 1`] = `
<LegayEditor
  classes={
    Object {
      "container": "LegayEditor-container-3",
      "editorContainer": "LegayEditor-editorContainer-5",
      "emojiSelect": "LegayEditor-emojiSelect-6",
      "footerSubsection": "LegayEditor-footerSubsection-2",
      "paper": "LegayEditor-paper-1",
      "readContainer": "LegayEditor-readContainer-4",
    }
  }
  readOnly={true}
  text="test 123"
/>
`;
